### DISCUSSION TITLE
POST https://xxxxxxxx.directus.app/graphql 400 - Error when fetching with public user

### DISCUSSION THREAD

**Discussion by Velpre** [directus/directus#16243] [Archive]
Hi 

I have a problem when I try to fetch data with public user. I am able to find what is causing the problem but can't find a solution for this. 

I have data_model called retro_games. Query i am using looks like this:
```
query MyQuery {
  retro_games_by_id(id: "2") {
      title
    description
    hashtags
    images
    user_created {
      first_name
      id
    }
  }
}
```
it is the user_created field that causes the problem but only when it is public user. If the user is logged in fetching works as it should. Also if i remove user_created i can fetch both when i am logged in and not. 

I can confirm that my public user have the same permissions as the user who is logged in. 
I am also sure that query is written correctly because I check it with "hasura cloud" and it returns data.

I can't understand why I can't get the user_created field when I fetch with public user but can get it if the user is logged in?

Thanks in advance

**Comment by rijkvanzanten** [comment]
> I can't understand why I can't get the user_created field when I fetch with public user but can get it if the user is logged in?

Sounds like a permissions problem to me. Does the public have read access to Users?

**Reply by Velpre** [reply]
You are right. I found that i forgot to grant permission specifically for users. Thank you very much :)
